Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-VolumeAffectedRowsStatefulAnomalyOnDatabase.yaml

```diff
@@ -0,0 +1,82 @@
+  'Goal: To detect anomalous data change/deletion. This query detects SQL queries that changed/deleted a large number of rows, which is significantly higher than normal for this database. This is a hunting query, so the training and the detection occur on the whole time window (controlled by 'queryPeriod' parameter). The user can set the minimal threshold for anomaly by changing the threshold parameters volThresholdZ and volThresholdQ (higher threshold will detect only more severe anomalies).'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureSql
+    dataTypes:
+      - AzureDiagnostics
+queryFrequency: 1h
+queryPeriod: 15d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
+relevantTechniques:
+  - T1485 
+  - T1565 
+  - T1491
+tags:
+  - SQL
+query: |
+    let volumeThresholdZ = 3.0;                     // Minimal threshold for the Zscore to trigger anomaly (number of standard deviations above mean). If set higher, only very significant alerts will fire.
+    let volumeThresholdQ = volumeThresholdZ;        // Minimal threshold for the Qscore to trigger anomaly (number of Inter-Percentile Ranges above high percentile). If set higher, only very significant alerts will fire.
+    let volumeThresholdHardcoded = 500;             // Minimal value for the volume metric to trigger anomaly.
+    let monitoredColumn = 'AffectedRows';           // The name of the column for volumetric anomalies.
+    let processedData = materialize (
+        AzureDiagnostics
+        | where Category == 'SQLSecurityAuditEvents' and action_id_s has_any ("RCM", "BCM") // Keep only SQL affected rows
+        | project TimeGenerated, PrincipalName = server_principal_name_s, ClientIp = client_ip_s, HostName = host_name_s, ResourceId,
+                  ApplicationName = application_name_s, ActionName = action_name_s, Database = strcat(LogicalServerName_s, '/', database_name_s),
+                  IsSuccess = succeeded_s, AffectedRows = affected_rows_d,
+                  ResponseRows = response_rows_d, Statement = statement_s, 
+                  Error = case( additional_information_s has 'error_code', toint(extract("<error_code>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+)", 1, additional_information_s))
+                        , 0),
+                  State = case( additional_information_s has 'error_state', toint(extract("<error_state>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+), Level ([0-9.]+)", 2, additional_information_s))
+                        , 0)
+        | extend QuantityColumn = column_ifexists(monitoredColumn, 0)
+        | sort by TimeGenerated desc
+        | extend RowNumber = row_number(),
+                             WindowType = case( TimeGenerated >= ago(detectionWindow), 'detection',
```

Comment:
detectionWindow doesn't appear to be a valid column or variable and is causing the query to fail.

Reply:
This also needs to be fixed still.